Unable to build with -Prelease due to javadoc error

##### Issue Overview

The current release build is failing if the release profile is enabled due to a javadoc error.

##### Expected Behaviour
The release builds

##### Current Behaviour
The build fails when generating the javadoc for the arquillian-core-api module:

```
1 error
[INFO] FAILURE build of project org.jboss.arquillian.core:arquillian-core-api
[INFO] Builder state: blocked=0 finished=1 ready-or-running=0 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.942 s (Wall Clock)
[INFO] Finished at: 2022-07-27T21:53:08-05:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:3.1.1:javadoc (default-cli) on project arquillian-core-api: An error has occurred in Javadoc report generation: 
[ERROR] Exit code: 1 - javadoc: warning - The old Doclet and Taglet APIs in the packages
[ERROR] com.sun.javadoc, com.sun.tools.doclets and their implementations
[ERROR] are planned to be removed in a future JDK release. These
[ERROR] components have been superseded by the new APIs in jdk.javadoc.doclet.
[ERROR] Users are strongly recommended to migrate to the new APIs.
[ERROR] javadoc: error - invalid flag: -Xdoclint:none
[ERROR] 
[ERROR] Command line was: /Library/Java/JavaVirtualMachines/jdk-11.0.13.jdk/Contents/Home/bin/javadoc @options @packages
[ERROR] 
[ERROR] Refer to the generated Javadoc files in '/Users/starksm/Dev/JBoss/Weld/arquillian-core/core/api/target/site/apidocs' dir.
[ERROR] 
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

##### Steps To Reproduce
1. cd arquillian-core/core/api
2. mvn -Prelease compile javadoc:javadoc

##### Additional Information

```
<details>
 <summary>$mvn --version</summary>
Apache Maven 3.8.4 (9b656c72d54e5bacbed989b64718c159fe39b537)
Maven home: /usr/local/Cellar/maven/3.8.4/libexec
Java version: 11.0.16, vendor: Eclipse Adoptium, runtime: /Library/Java/JavaVirtualMachines/temurin-11.jdk/Contents/Home
Default locale: en_US, platform encoding: UTF-8
OS name: "mac os x", version: "12.5", arch: "x86_64", family: "mac"
</details>
```
I am seeing the same problem on another linux build system as well. The full output of the mvn -X command and options file passed to the javadoc command is attached in the build.log file.

[build.log](https://github.com/arquillian/arquillian-core/files/9205137/build.log)

--- a/pom.xml
+++ b/pom.xml
@@ -23,11 +23,9 @@
 
   <properties>
     <!-- jboss-parent override -->
-    <!-- 3.8.1-jboss-1 is causing issues with test deps -->
-    <version.compiler.plugin>3.8.0-jboss-2</version.compiler.plugin>
-
     <maven.compiler.target>1.8</maven.compiler.target>
     <maven.compiler.source>1.8</maven.compiler.source>
+    <version.javadoc.plugin>3.4.0</version.javadoc.plugin>
 
     <!-- External Deps -->
     <version.shrinkwrap_shrinkwrap>1.2.6</version.shrinkwrap_shrinkwrap>
@@ -128,11 +126,11 @@
           <plugin>
             <artifactId>maven-javadoc-plugin</artifactId>
             <configuration>
-              <doclet>org.jboss.apiviz.APIviz</doclet>
+              <doclet>nl.talsmasoftware.umldoclet.UMLDoclet</doclet>
               <docletArtifact>
-                <groupId>org.jboss.apiviz</groupId>
-                <artifactId>apiviz</artifactId>
-                <version>1.3.2.GA</version>
+                <groupId>nl.talsmasoftware</groupId>
+                <artifactId>umldoclet</artifactId>
+                <version>2.0.18</version>
               </docletArtifact>
               <useStandardDocletOptions>true</useStandardDocletOptions>
               <charset>UTF-8</charset>

